CannibalisticZombies/Assets/Scripts/Interactable/Door.cs

```diff
@@ -21,6 +19,38 @@ public class Door : InteractionObject
 
         /// this is a private boolean that checks if the door is either closed or open, we assume its closed at the start  
         private bool doorOpen = false;
+
+        /// pivot of the door that will be rotated in the script
+        [SerializeField] 
+        private GameObject pivot;
+
+
+        ///  closedRotation that the door is placed
+        private Quaternion closedRotation;
+
+        /// when facing away from the door
+        private Quaternion openRotationAway;
+
+        /// when facing towards to the open
+        private Quaternion openRotationTowards;
+
+        /// the rotation angle of the door
+        private float rotateAngle = 90f; 
+
+        ///  the speed of the door closing/opening
+        [SerializeField]
+        private float doorSpeed = 50f;
+
+        ///  stored coroutine to avoid multiple coroutines running at once
+        private Coroutine storedCoroutine;
```

Comment:
Rename the coroutine to something that has more meaning, like `rotateDoorCoroutine` so that if other people read the code they will instantly know what it is for